extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreator.java

```diff
@@ -168,15 +158,23 @@ private YoutubeDashManifestCreator() {
      * Exception that is thrown when the {@link YoutubeDashManifestCreator} encounters a problem
      * while creating a manifest.
      */
-    public static final class YoutubeDashManifestCreationException extends Exception {
+    public static final class CreationException extends Exception {
 
-        YoutubeDashManifestCreationException(final String message) {
+        CreationException(final String message) {
             super(message);
         }
 
-        YoutubeDashManifestCreationException(final String message, final Exception e) {
+        CreationException(final String message, final Exception e) {
             super(message, e);
         }
+
+        public static CreationException couldNotAdd(final String element, final Exception e) {
+            return new CreationException("Could not add " + element + " element", e);
+        }
+
+        public static CreationException couldNotAdd(final String element, final String reason) {
+            return new CreationException("Could not add " + element + " element: " + reason);
+        }
```

Comment:
@Stypox ^

Reply:
I mean, they are just a way to build the exception with a default message. Making them constructors would not be possible because of conflicting signatures (and also because I think `couldNotAdd` is more understandable). I didn't create a separate exception because this is really just yet another internal error that may happen, so there is no need to expose to the user another type of exception. The reason I created these handy builders is just to deduplicate some "Could not add" strings.

Reply:
Sounds valid. Please document that these are shortcuts in the code I think we are fine then.

Reply:
Done